### DISCUSSION TITLE
Could not deploy to Heroku

### DISCUSSION THREAD

**Discussion by omaga-w** [aahnik/tgcf#106] [Q&A] [answered]
please who deployed on Heroku, i deployed mine on heroku but its not working

**Comment by ghost** [comment]
In my case it works very well. (live mode)


When the build is complete, Click 'Configure Dynos' and turn on tgcf app.

**Reply by omaga-w** [reply]
i turned on the app but not working, please are on telegram is we an discuss

**Reply by omaga-w** [reply]
i'm thinking it is configuration

**Reply by ghost** [reply]
Sorry, I don't understand what you're saying. Please explain in more detail.

**Reply by omaga-w** [reply]
I mean your username let chat on telegram

**Reply by omaga-w** [reply]
Please

**Reply by ghost** [reply]
Why should we talk on Telegram? I don't understand.

**Reply by omaga-w** [reply]
I said I turned on the app but still not working
please give me your telegram username to we can chat and show you what I have done please

**Reply by omaga-w** [reply]
I'm suspecting my configuration is the problem let me send it so you see if it wrong

**Reply by ankit-sinha-18** [reply]
Check your heroku logs

**Reply by omaga-w** [reply]
Yes I did it giving "struct.error: unpack requires buffer of 275 byte":

**Reply by ankit-sinha-18** [reply]
You have not generated string session

**Comment by aahnik** [comment, marked as answer]
Please read this page fully and carefully https://github.com/aahnik/tgcf/wiki/Deploy-to-Heroku

Every step is explained with details + screenshots

**Reply by mujahidazamcsm** [reply]
Hi.. 
I am deploying to heroku and followed all steps in the wiki. 

using bot_token
created the session string
entered the `TGCF_CONFIG` as below

*********************
```yaml
forwards:
  - source: 1427091357
    dest: [1262983491,1392233102]


show_forwarded_from: false
```

****************************************

also tried the group/channel id as below (i.e. prefixing "-100" to the ids)
```yaml
 - source: -1001427091357
    dest: [-1001262983491,-1001392233102]
```

but the messages are not getting forwarded.. I tried restarting the dynos as well.. below is the log.. can u plz guide me on next steps

```shell
2021-05-05T09:18:48.834746+00:00 heroku[worker.1]: Restarting
2021-05-05T09:18:48.839181+00:00 heroku[worker.1]: State changed from up to starting
2021-05-05T09:18:50.026492+00:00 heroku[worker.1]: Stopping all processes with SIGTERM
2021-05-05T09:18:50.154899+00:00 heroku[worker.1]: Process exited with status 143
2021-05-05T09:19:13.311160+00:00 heroku[worker.1]: Starting process with command `tgcf`
2021-05-05T09:19:14.136243+00:00 heroku[worker.1]: State changed from starting to up
```

**Reply by ankit-sinha-18** [reply]
Generate session string using phone number

**Reply by mujahidazamcsm** [reply]
So on "repl.it", give the phone num instead of the bot_token.. right?

**Reply by ankit-sinha-18** [reply]
Yes

**Reply by mujahidazamcsm** [reply]
trying now

**Reply by mujahidazamcsm** [reply]
thanks Ankit.. got it working.. thanks for the quick response...

should we look at updating the documentation to mention to use phone instead of bot

**Reply by aahnik** [reply]
hi @mujahidazamcsm 

You can use either phone no or bot token.

Phone no is for using tgcf with your user account. Bot token is for using a bot.

If you have used a bot, you must add the bot to the groups and channels. If you can't do that, use user account.

Bots privacy mode should be off. That means bot should be able to listen all messages in the groups or channels it is present in.


--------

i will add these details to the wiki! and also video is coming!

**Reply by mujahidazamcsm** [reply]
Thanks Aahnik!

**Reply by aahnik** [reply]
![image](https://user-images.githubusercontent.com/66209958/117131023-7ef04880-adbe-11eb-85c7-7936706b7ed1.png)

**Reply by mujahidazamcsm** [reply]
Hi Aahnik,

So far working fine with older groups and channels which have 10digits in their groupid 

I created multiple new groups today and its groupid has only 9digits and the fwdng doesnt seem to work.. 
I have tried with prefixing -100 to the groupid.. and also without the prefixing -100.

Is that supposed to be handled differently?

Created new channels and that seems to work as it has id with 10digits

**Reply by aahnik** [reply]
hi @mujahidazamcsm, after this #118 is solved, you can use usernames/links/phone-numbers always.

Currently, you may use username/link/phone number, when running in past mode. And also in the destination for the live mode.

To get the chat id of a chant (channel/group/personal chat), you must use the [userinfobot](https://telegram.me/userinfobot)

**Reply by mujahidazamcsm** [reply]
thanks for the update Aahnik.. I will chk that out. btw I was using above with live mode.. where it was failing
